dotnet-testing/MockingWithNSubstitute/MockingWithNSubstitute/NotificationService.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace MockingWithNSubstitute
+{
+    public class NotificationService : INotificationService
+    {
+        private readonly IEmailService _emailService;
+
+        public NotificationService(IEmailService emailService)
+        {
+            _emailService = emailService;
+        }
+
+        public bool NotifyUser(User user, string message)
+        {
+            if (user?.Email is null ||
```

Comment:
Since this is nullable enabled, we are guaranteed that `user` is not null. The same applies to `user.Email` as the User record also does not allow null for `Email`

Reply:
Good point, sorted.